ci/README.md

```diff
@@ -48,12 +48,58 @@ following:
 
 ## `check_enterprise_merge.sh`
 
-When you open a PR on community, if it creates a conflict with
-enterprise-master, the check-merge-to-enterprise will fail. Say your branch name
-is `$PR_BRANCH`, we will refer to `$PR_BRANCH` on community as
-`community/$PR_BRANCH` and on enterprise as `enterprise/$PR_BRANCH`. If the
-job already passes, you are done, nothing further required! Otherwise follow the
-below steps. First make sure these two things are the case:
+This check exists to make sure that we can always merge the `master` branch of
+`community` into the `enterprise-master` branch of the `enterprise` repo.
+There are two conditions in which this check passes:
+1. There are no merge conflicts between your PR branch and `enterprise-master`.
+2. There are merge conflicts, but there is a branch with the same name in the
+   enterprise repo that:
+   1. Contains the last commit of the community branch with the same name.
+   2. Merges cleanly into `enterprise-master`
+
+If the job already passes, you are done, nothing further required! Otherwise
+follow the below steps.
+
+### Prerequisites
```

Comment:
Why is this a prerequisite? This seems more of a nice-to-have

Reply:
(I assume you mean the `git rerere` part here, not the `community` remote.) I put it as a prerequisite, because if you enable it after the fact it doesn't work. Since it only tracks merges when it's on.